flash_examples/integrations/learn2learn/image_classification_imagenette_mini.py

```diff
@@ -14,71 +14,139 @@
+
+"""
+
 import warnings
+from dataclasses import dataclass
+from typing import Callable, Tuple, Union
 
 import kornia.augmentation as Ka
 import kornia.geometry as Kg
 import learn2learn as l2l
+import numpy as np
 import torch
 import torchvision
+import torchvision.transforms as T
+from PIL import Image
 from torch import nn
 
 import flash
 from flash.core.data.io.input import DataKeys
+from flash.core.data.io.input_transform import InputTransform
 from flash.core.data.transforms import ApplyToKeys, kornia_collate
 from flash.image import ImageClassificationData, ImageClassifier
 
 warnings.simplefilter("ignore")
 
 # download MiniImagenet
-train_dataset = l2l.vision.datasets.MiniImagenet(root="data", mode="train", download=True)
-val_dataset = l2l.vision.datasets.MiniImagenet(root="data", mode="validation", download=True)
+train_dataset = l2l.vision.datasets.MiniImagenet(root="./", mode="train", download=False)
+val_dataset = l2l.vision.datasets.MiniImagenet(root="./", mode="validation", download=False)
+
+
+@dataclass
+class ImageClassificationInputTransform(InputTransform):
 
-transform = {
-    "per_sample_transform": nn.Sequential(
-        ApplyToKeys(
+    image_size: Tuple[int, int] = (196, 196)
+    mean: Union[float, Tuple[float, float, float]] = (0.485, 0.456, 0.406)
+    std: Union[float, Tuple[float, float, float]] = (0.229, 0.224, 0.225)
+
+    def per_sample_transform(self):
+        return T.Compose(
+            [
+                ApplyToKeys(
+                    DataKeys.INPUT,
+                    T.Compose(
+                        [
+                            T.ToTensor(),
+                            Kg.Resize((196, 196)),
+                            # SPATIAL
+                            Ka.RandomHorizontalFlip(p=0.25),
+                            Ka.RandomRotation(degrees=90.0, p=0.25),
+                            Ka.RandomAffine(degrees=1 * 5.0, shear=1 / 5, translate=1 / 20, p=0.25),
+                            Ka.RandomPerspective(distortion_scale=1 / 25, p=0.25),
+                            # PIXEL-LEVEL
+                            Ka.ColorJitter(brightness=1 / 30, p=0.25),  # brightness
+                            Ka.ColorJitter(saturation=1 / 30, p=0.25),  # saturation
+                            Ka.ColorJitter(contrast=1 / 30, p=0.25),  # contrast
+                            Ka.ColorJitter(hue=1 / 30, p=0.25),  # hue
+                            Ka.RandomMotionBlur(kernel_size=2 * (4 // 3) + 1, angle=1, direction=1.0, p=0.25),
+                            Ka.RandomErasing(scale=(1 / 100, 1 / 50), ratio=(1 / 20, 1), p=0.25),
+                        ]
+                    ),
+                ),
+                ApplyToKeys(DataKeys.TARGET, torch.as_tensor),
+            ]
+        )
+
+    def train_per_sample_transform(self):
+        return T.Compose(
+            [
+                ApplyToKeys(
+                    "input",
```

Comment:
minor nit for consistency:

```suggestion
                    DataKeys.INPUT,
```